Sources/NIOCore/ByteBuffer-views.swift

```diff
@@ -60,6 +60,13 @@ public struct ByteBufferView: RandomAccessCollection {
         return i + 1
     }
 
+    @inlinable
+    public var count: Int {
+        // Unchecked is safe here: Range enforces that upperBound is strictly greater than
+        // lower bound. Annoyingly, Range itself does not do this.
```

Comment:
is this worth a bug report or even directly a patch?

Reply:
Yes, probably.

Reply:
Surely it's not safe for Range to do that? e.g. `-1 ... Int.max`?

Reply:
Yup good catch, that will trap.

Reply:
I would then suggest to remove `Annoyingly, Range itself does not do this.` and add that we guarantee `range.lowerBound >= 0`.